ryoanji/src/ryoanji/nbody/direct.cuh

```diff
@@ -99,13 +101,14 @@ __global__ void directKernel(int numSource, const T* __restrict__ x, const T* __
 }
 
 template<class T>
-void directSum(std::size_t numBodies, const T* x, const T* y, const T* z, const T* m, const T* h, T* p, T* ax, T* ay,
-               T* az)
+void directSum(size_t first, size_t last, size_t numBodies, const T* x, const T* y, const T* z, const T* m, const T* h,
```

Comment:
isn't numBodies always last-first? maybe we don't need to pass that.

Reply:
For the direct sum we need it. `last - first` is the number of targets, but `numBodies` is the number of sources here, so they're different.